### PR TITLE
[WIP] Tracing PoC

### PR DESCRIPTION
**Pull request by sbueringer**
**What this PR does / why we need it**:
Please ignore for now 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: bootstrap/kubeadm/main.go [hunk 1]
```diff
@@ -166,6 +179,8 @@ func main() {
 		}()
 	}
 
+	tp := traceutil.NewProvider(tracingEndpoint, tracingSamplingRatePerMillion, "capi-kubeadm-bootstrap-controller-manager")
```
Comment (apricote) [inline bootstrap/kubeadm/main.go:182]: Would it make sense to align the name with the user agent 2 lines below? (Question also applies to the other 3 controllers)

Comment (sbueringer) [reply bootstrap/kubeadm/main.go:182]: Yes that would make sense.

Just fyi, this PR is currently only an experiment. I don't know when/if we will add tracing and it's entirely possible that we will add this feature to controller-runtime so we only have to enable it in CAPI

**Comment by vincepri** [comment]
@sbueringer can this be closed?

**Comment by sbueringer** [comment]
I moved this PR to my fork. I don't really have time to work on it now or any time soon.

/close

**Pull request closed without merging**